core/src/main/scala/io/finch/ToService.scala

```diff
@@ -91,12 +93,13 @@ object ToService {
       }
   }
 
-  implicit def hlistTS[A, EH <: Endpoint[A], ET <: HList, CTH, CTT <: HList](implicit
+  implicit def hlistTS[F[_], A, EH <: Endpoint[F, A], ET <: HList, CTH, CTT <: HList](implicit
     ntrA: ToResponse.Negotiable[A, CTH],
     ntrE: ToResponse.Negotiable[Exception, CTH],
+    effect: Effect[F],
```

Comment:
I'm thinking we should/might? require `ConcurrentEffect` as Finagle can cancel its promises/futures so we can propagate that back to the users' code. This would definitely shrink the space of effective implementation we can adapt so we should be really careful with our reasoning here.

Reply:
I see the reason but from the top of my head there are only two async implementations available that satisfy `ConcurrentEffect` laws: `cats.effect.IO` and `monix.Task`. Neither `Rerunnable` nor `Arrow` support it

Reply:
I feel it would be great if we could still support cancellable endpoints. Maybe there is way to distinguish those (cancellable vs. non-cancelable) in `Bootstrap`?

Again, this falls outside of the scope of this PR. I agree with you starting with just simple `Effect` is totally reasonable.

Reply:
Yeah, that could a one way to solve this. I think it's totally fine if we just focus on `Effect` for now and figure out cancellations later.